src/main/java/org/testng/internal/ClassHelper.java

```diff
@@ -165,16 +166,26 @@ private static void logClassNotFoundError(String className, Exception ex) {
    * the Java access rules).
    */
   public static Set<Method> getAvailableMethods(Class<?> clazz) {
-    Set<Method> methods = Sets.newHashSet();
-    methods.addAll(Arrays.asList(clazz.getDeclaredMethods()));
+    Map<String, Set<Method>> methods = Maps.newHashMap();
+    for (final Method method : clazz.getDeclaredMethods()) {
+      if (methods.containsKey(method.getName())) {
+        methods.get(method.getName()).add(method);
+        continue;
+      }
+      methods.put(method.getName(), new HashSet<Method>() {{add(method);}});
```

Comment:
Nitpick: I'd prefer reversing the logic and not use `continue`:

    String name = method.getName();
    Set<Method> methods = methodsByNames.get(name);
    if (methods == null) {
        methods = new HashSet<>();
        methodsByNames.put(name, methods);
    }
    methods.add(method);